python/ray/monitor.py

```diff
@@ -488,6 +492,106 @@ def driver_removed_handler(self, unused_channel, data):
 
         self._clean_up_entries_for_driver(driver_id)
 
+    def _xray_clean_up_entries_for_driver(self, driver_id):
+        """Remove this driver's object/task entries from redis.
+
+        Removes control-state entries of all tasks and task return
+        objects belonging to the driver.
+
+        Args:
+            driver_id: The driver id.
+        """
+
+        xray_task_table_prefix = (
+            ray.gcs_utils.TablePrefix_RAYLET_TASK_string.encode("ascii"))
+        xray_object_table_prefix = (
+            ray.gcs_utils.TablePrefix_OBJECT_string.encode("ascii"))
+
+        task_table_infos = {}  # task id -> TaskInfo
+        task_table_objects = self.state.task_table()
+        driver_id_hex = binary_to_hex(driver_id)
+        for task_id_hex in task_table_objects:
+            if len(task_table_objects[task_id_hex]) == 0:
```

Comment:
instead of indexing into `task_table_objects` here, let's change the for loop to be `for task_id_hex, task_info in self.state.task_table().items()` or something like that